src/sql/parts/notebook/outputs/tableRenderers.ts

```diff
@@ -64,6 +66,7 @@ export function renderDataResource(
 	// Set the height dynamically if the grid's height is < 500px high; otherwise, set height to 500px
 	tableContainer.style.height = rowsHeight >= 500 ? '500px' : rowsHeight.toString() + 'px';
 
+	attachTableStyler(detailTable, options.themeService);
```

Comment:
Have you tested hitting this code. Closing the notebook then changing the theme? You might have errors thrown since you should dispose of the listener when the notebook is destroyed.

Reply:
Hmm, I don't see any obvious exceptions being thrown or any errors pop up. Let me see about disposing the listener just to be cautious though.

Reply:
O that would be because you aren't disposing the table either...you might have a pretty serious memory leak here.

Reply:
It seems to me that this function should return a diposable rather than undefined, that disposes of everything this function creates. Presumably this function is being called by classes that can handle disposing.

Reply:
Will be resolved in an upcoming PR when we create angular components for the renderers.